From a97dd36dfbca0f6e1f2bc9e26c39b687058871d1 Mon Sep 17 00:00:00 2001
From: Tonyliu <Bo.Liu@windriver.com>
Date: Fri, 19 Jun 2009 20:05:00 +0800
Subject: [PATCH] mxc_nand: add mtd-nand partitions support

Adding mtd-nand partitions support. If the parser failed to find
the partitions automatically, It will try with a pre-defined
partitions map defined in platform-device level.

Signed-off-by: Tonyliu <Bo.Liu@windriver.com>
---
 arch/arm/mach-mx2/mx27ads.c |   32 ++++++++++++++++++++++++++++++++
 1 files changed, 32 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx2/mx27ads.c b/arch/arm/mach-mx2/mx27ads.c
index 291956d..1385fe2 100644
--- a/arch/arm/mach-mx2/mx27ads.c
+++ b/arch/arm/mach-mx2/mx27ads.c
@@ -177,9 +177,41 @@ EXPORT_SYMBOL(gpio_keypad_inactive);
 #endif /* CONFIG_MXC_KEYBOARD */
 
 #ifdef CONFIG_MTD_NAND
+static struct mtd_partition mxc_nand_partitions[] = {
+	{
+	 .name = "IPL-SPL",
+	 .size = 128 * 1024,
+	 .offset = 0x00000000,
+	 .mask_flags = MTD_WRITEABLE	/* force read-only */
+	},
+	{
+	 .name = "nand.Kernel",
+	 .size = 4 * 1024 * 1024,
+	 .offset = MTDPART_OFS_APPEND,
+	 .mask_flags = 0},
+	{
+	 .name = "nand.rootfs",
+	 .size = 22 * 1024 * 1024,
+	 .offset = MTDPART_OFS_APPEND,
+	 .mask_flags = 0},
+	{
+	 .name = "nand.userfs",
+	 .size = MTDPART_SIZ_FULL,
+	 .offset = MTDPART_OFS_APPEND,
+	 .mask_flags = 0},
+};
+
+static struct flash_platform_data mxc_nand_data[1] = {
+	{
+	 .parts = mxc_nand_partitions,
+	 .nr_parts = ARRAY_SIZE(mxc_nand_partitions),
+	}
+};
+
 static struct mxc_nand_platform_data mx27ads_nand_board_info = {
 	.width = 1,
 	.hw_ecc = 1,
+	.flash_data = mxc_nand_data,
 };
 #endif
 
-- 
1.5.5.1

